### PR TITLE
[feature] TravisCI now allows per component configuration + allows to set a specific make command to run

### PR DESCRIPTION
**Pull request by edulop91**
### Summary
- Mostly lifted from the Atlantis implementation

### Test Plan
- unittest
- run in existing repo

### CODE REVIEW

File: plan/plan.go [hunk 1]
```diff
@@ -47,6 +48,14 @@ type AtlantisComponent struct {
 	RolePath string `yaml:"role_path"`
 }
 
+type TravisComponent struct {
+	Enabled        bool `yaml:"enabled"`
```
Comment (edulop91) [inline plan/plan.go]:
```suggestion
	Enabled        bool
```

File: plan/travisci.go [hunk 2]
```diff
@@ -1,83 +1,108 @@
...
 	FoggVersion string
-	TestBuckets [][]string
+	TestBuckets [][]TravisProject
+	AWSProfiles map[string]AWSRole
 }
 
-func (p *Plan) buildTravisCI(c *v2.Config, version string) TravisCI {
-	if p.Accounts == nil {
-		panic("buildTravisCI must be run after buildAccounts")
+func (p *Plan) buildTravisCI(c *v2.Config, foggVersion string) TravisCI {
```
Comment (edulop91) [inline plan/travisci.go]:
```suggestion
// TODO(el): mostly a duplicate of buildAtlantis(). refactor later
func (p *Plan) buildTravisCI(c *v2.Config, foggVersion string) TravisCI {
```

File: config/v1/config.go [hunk 3]
```diff
@@ -126,9 +126,10 @@ type Module struct {
 }
 
 type TravisCI struct {
-	Enabled        bool   `json:"enabled,omitempty"`
-	AWSIAMRoleName string `json:"aws_iam_role_name"`
-	TestBuckets    int    `json:"test_buckets"`
+	Enabled        *bool   `json:"enabled,omitempty"`
+	AWSIAMRoleName *string `json:"aws_iam_role_name"`
```
Comment (ryanking) [inline config/v1/config.go]: probably omitempty the rest of these

File: plan/travisci.go [hunk 4]
```diff
@@ -1,83 +1,109 @@
...
+			proj := TravisProject{
+				Name:    fmt.Sprintf("accounts/%s", name),
+				Dir:     fmt.Sprintf("terraform/accounts/%s", name),
+				Command: acct.TravisCI.Command,
+			}
+
+			projects = append(projects, proj)
+
+			// Grab all profiles from accounts
+			awsProfiles[acct.Providers.AWS.Profile] = AWSRole{
```
Comment (ryanking) [inline plan/travisci.go]: what about backend?

**Pull request merged**